clients/web/src/views/widgets/SearchFieldWidget.js

```diff
@@ -253,6 +258,57 @@ var SearchFieldWidget = View.extend({
             }
         }, this));
     }
+}, {
+    _allowedSearchMode: {},
+
+    addMode: function (mode, types, description, help) {
+        if (!SearchFieldWidget._allowedSearchMode[mode]) {
```

Comment:
This is minor, but in my opinion, it's probably clearer style to check whether the `mode` already exists [using `_.has`](http://underscorejs.org/#has), and if so, throw the error within the body of the `if`. Then, do the normal work of registering the mode in the body of the function (not inside an `if` or `else`).

I think this is preferable because it tests any preconditions at the start of the function call, then moves into the function logic with the assumption that they're true. This allows all the logic to be at the top (non-indented) level of the function, and keeps the error handling (throw an exception) close to the error check (the condition).